public/docs/ts/latest/guide/component-communication.jade

```diff
@@ -0,0 +1,707 @@
+include ../../../../_includes/_util-fns
+
+:marked
+  ## Component Communication
+
+  Real applications rarely use a single component to implement the entire functionality of the app. To create structured, testable 
+  and maintainable applications, we'd better decompose the entire logic into a set of loosely-coupled (or decoupled) components.
+  This way, we can create smaller building blocks with their own boundaries of responsibility, and compose the entire application
+  from them.
+  
+  Generally, an Angular app is a tree of components &mdash; with the bootstrapped component in its root. Thinking of an app's 
+  structure as a tree, there are components that are in the same branch of the component tree &mdash; these form parent and child hierarchies
+  &mdash;, and components in different branches &mdash; they are not directly related. Evidently, components need to communicate with each other
+  to carry out the tasks the app is created for.
+  
+  As we already learned, _component_ is a pivotal concept of Angular, and it was designed with _inter-component communication_ in mind.
+  We can think of the following information flows:
+  * From a parent to its children
+  * From a child to its parent
+  * Among unrelated, completely decoupled components
+  
+  In this chapter, we will learn six frequently used Angular mechanisms and techniques available for inter-component communication.
+  We will also get acquainted with the tradeoffs of each mechanism, and understand the scenarios best suited for using a particular method of communication.
+  Before diving deeply into the details, here is a short overview of these techniques:
+  
+  __#1: Parent to child communication with input property binding__
+  
+  The `prop-value` expression of the parent is bound to the `child-prop` input property of the child. `child-prop` is updated every time when `prop-value` changes:
+  
++makeExample('component-communication/ts/fragments/property-binding-intro.html')
+  
+:marked
+  We can bind multiple properties of a single child component to the parent, and, implicitly, we can bind multiple children to the parent &mdash; 
+  ie. with `ng-for`:
+  
++makeExample('component-communication/ts/fragments/property-binding-intro-2.html')
+  
+:marked
+  [Jump to details](#-1-parent-to-child-communication-with-input-property-binding)
+
+:marked
+  __#2: Child to parent communication with event emitting__
+  
+  When a child needs to communicate with its parent, it can emit an event. The parent can define  its own event handler method , and bind this method 
+  to the child's output property:
```

Comment:
Confusing. The reader doesn't know what an `output property` is yet or how it relates to the `child-event` property.

I feel this example is confusing because of the `<button>`. We can trim it down to just
```
parent-comp>
  <child-comp (child-event)='handleMyEvent($event)'>
  </child-comp>
</parent-comp>
```

HEADS UP: Must rename all kebab-case sample component input/output properties (`child-event`) to camelCase (`childEvent`).